File: sem-3-labs/dsal/lab2/q2.c
#include <stdio.h>
#include <stdlib.h>
int Smallest(int *arr, int l)
{
    int s = *arr;
    for (int i = 1; i < l; i++)
    {
        if (*(arr + i) < s)
            s = *(arr + i);
    }
    return s;
}
void main()
{
    int l;
    printf("enter length of array ");
    scanf("%d", &l);
    int *arr = (int *)malloc(l * sizeof(int));
    printf("enter numbers\n");
    for (int i = 0; i < l; i++)
        scanf("%d", &arr[i]);
    int s = Smallest(arr, l);
    printf("the smallest number is: %d", s);
}

File: sem-6-labs/itl/lab4/q2.py
class q2:
	def q2(self, arr, key):
		for i in range(0, len(arr)):
			for j in range(i+1, len(arr)):
				if arr[i] + arr[j] == key:
					print(i+1, j+1, "\n")

n = int(input("enter number of elements : "))
arr = list(map(int, input("\nenter the numbers: ").strip().split()))[:n]
key = int(input("enter target value : "))

q2().q2(arr, key)